### ISSUE TITLE
Issues with class A networks

### ISSUE THREAD

**Issue by davidchappelle** [arvidn/libtorrent#34]
For testing purposes we have hosted our own DHT bootstrap server. To facilitate introduction to the DHT we have also setup a handful of bot clients using https://github.com/davidchappelle/bootstrap-dht-bot. Our bootstrap server is hosted by Digital Ocean and was assigned a public IP address as 45.x.y.z. Surprisingly, we discovered that our DHT was not functioning correctly. Upon investigation we found this:

**libtorrent-rasterbar/src/kademlia/dht_tracker.cpp**
```
        if (m_settings.ignore_dark_internet && ep.address().is_v4())
        {
            address_v4::bytes_type b = ep.address().to_v4().to_bytes();

            // these are class A networks not available to the public
            // if we receive messages from here, that seems suspicious
            boost::uint8_t class_a[] = { 3, 6, 7, 9, 11, 19, 21, 22, 25
                , 26, 28, 29, 30, 33, 34, 45, 48, 51, 52, 56, 102, 104 };

            int num = sizeof(class_a)/sizeof(class_a[0]);
            if (std::find(class_a, class_a + num, b[0]) != class_a + num)
                return true;
        }
```

Since our Digital Ocean server was assigned a 45.x.y.z address and ignore_dark_internet is enabled by default, our DHT was rendered useless. Is this perhaps a legacy issue with these class A networks? Does this list of IP addresses need to be reviewed and updated? Clearly we can use the setting to work around this but it was a bit painful arriving here.

**Comment by davidchappelle** [comment]
Looks like some of these have been re-purposed or re-assigned over the years:

https://en.wikipedia.org/wiki/List_of_assigned_/8_IPv4_address_blocks

**Comment by arvidn** [comment]
I will update this list in RC_1_0 and release it with the next stable. thanks!

**Comment by davidchappelle** [comment]
Seems to be fixed by commit f85d62b601b4c52675afa626436810f90c124ab6 on master as well as commit 0f2f14484f7ba2e9c0e15649339d2044acbcd233 on RC_1_0.
